lib/src/app/cast.js

```diff
@@ -18,58 +20,73 @@ import '../styles/player.css';
-        if (castPlayer.OOPlayer != null) {
-            castPlayer.notifySenders(castPlayer.OOPlayer.getState());
+    mediaManager.onLoad = (() => {
+        mediaManager.origOnLoad = mediaManager.onLoad;
+        return (event) => {
+            logger.info('MediaManager:onLoad', event);
+            const isAutoplayDesired = event && event.data && event.data.autoplay;
+            if (isAutoplayDesired) {
+                Timer.setIdle();
+            }
+            castPlayer.setAsset(event.data.media.customData);
+            if (castPlayer.OOPlayer != null) {
+                const state = castPlayer.OOPlayer.getState();
+                castPlayer.notifySenders(state);
+                if (isAutoplayDesired) {
+                    castPlayer.play();
+                }
+            }
+            mediaManager.origOnLoad(event);
         }
-        mediaManager.origOnLoad(event);
-    };
-
-    // backup of the original onPause event handler to use it after our custom logic
-    mediaManager.origPause = mediaManager.onPause;
+    })();
 
     /**
-     * onPause overrides the default behaviour to setup the default timeout when an assets it's getting paused
+     * Exotic class method extending by Google devs
+     * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
+     * onPause overrides the default behaviour to setup the default timeout
+     * when an assets it's getting paused
+     * @param {cast.receiver.MediaManager.Event} event
      */
-    mediaManager.onPause = function(e) {
-        logger.info('MediaManager:onPause', e);
-        Timer.setIdle(Timer.TIMEOUT.PAUSED);
-        mediaManager.origPause(e);
-    };
-
-    // backup of the original onPlay event handler to use it after our custom logic
-    mediaManager.origPlay = mediaManager.onPlay;
+    mediaManager.onPause = (() => {
+        mediaManager.origOnPause = mediaManager.onPause;
+        return (event) => {
+            logger.info('MediaManager:onPause', event);
+            Timer.setIdle(Timer.TIMEOUT.PAUSED);
+            if (castPlayer.OOPlayer != null) {
+                castPlayer.notifySenders(castPlayer.OOPlayer.getState());
+            }
+            mediaManager.origOnPause(event);
+        }
+    })();
 
     /**
+     * Exotic class method extending by Google devs
+     * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
      * onPlay override the original logic for play just to clear the timeout, if one was created
+     * @param {cast.receiver.MediaManager.Event} event
      */
-    mediaManager.onPlay = function(e) {
-        logger.info('MediaManager:onPlay', e);
-        Timer.setIdle();
-        mediaManager.origPlay(e);
-    };
+    mediaManager.onPlay = (() => {
+        mediaManager.origOnPlay = mediaManager.onPlay;
+        return (event) => {
+            logger.info('MediaManager:onPlay', event);
+            Timer.setIdle();
+            mediaManager.origOnPlay(event);
```

Comment:
Why this method does not have the same logic to notify the state to the senders?

Reply:
Have the same question. Should be addressed to author of this part:
@aldigun in this commit:
https://github.com/ooyala/chromecast-sample-receiver/commit/4e864e7468d05cbd44907aa85ea6f30171c2f677